roles/pulp_api/tasks/generate_database_fields_key.yml

```diff
@@ -28,7 +28,8 @@
   shell: >-
     set -o pipefail &&
     umask 0077 &&
-    openssl rand -base64 32 > {{ __pulp_db_fields_key_path }}
+    openssl rand -base64 32
+    | tr '+/' '-_' > {{ __pulp_db_fields_key_path }}
```

Comment:
Thank you very much for this.

I suspect that newer versions of fernet overcome this limitation, but the docs are outdated.

Reply:
You're welcome!

I originally read from the docs that this is necessary, later felt silly when I looked at the code and realized that it's not necessary in the current implementation, and then decided (or convinced myself?) that it's still better to interpret the docs as representing the true design intent while the particular implementation is more likely to change.

Reply:
Researching it a bit more, it turns out that `cryptography.fernet` is a particular implementation of an external spec, defined at: https://github.com/fernet/spec/blob/master/Spec.md

The spec explicitly states that `base64url` should always be used.